src/main/scala/mesosphere/marathon/core/storage/store/impl/zk/RichCuratorFramework.scala

```diff
@@ -159,22 +158,25 @@ class RichCuratorFramework(val client: CuratorFramework) {
     */
   @SuppressWarnings(Array("CatchFatal"))
   def blockUntilConnected(lifecycleState: LifecycleState): Unit = {
-    val timeoutAt: Long = System.currentTimeMillis() + client.getZookeeperClient.getConnectionTimeoutMs
+    val zkTimeout: Int = client.getZookeeperClient.getConnectionTimeoutMs
```

Comment:
Again - why is the timeout coming from the client and not from `zk.config`?

Reply:
Because `zk.config` is used at the time of client creation.